tests/components/azure_data_explorer/test_config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Test the AEH config flow."""
+from azure.kusto.data.exceptions import KustoAuthenticationError, KustoServiceError
+import pytest
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.azure_data_explorer.const import DOMAIN
+
+from .const import BASE_CONFIG, UPDATE_OPTIONS
+
+
+async def test_form(hass, mock_setup_entry) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}, data=None
+    )
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["errors"] == {}
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        BASE_CONFIG.copy(),
+    )
+
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result2["title"] == "cluster.region.kusto.windows.net"
+    mock_setup_entry.assert_called_once()
+
+
+@pytest.mark.parametrize(
+    ("test_input", "expected"),
+    [
+        (KustoServiceError("test"), "cannot_connect"),
+        (KustoAuthenticationError("test", Exception), "invalid_auth"),
+        (Exception(), "unknown"),
+    ],
+)
+async def test_connection_error(
+    test_input,
+    expected,
+    hass,
+    mock_execute_query,
+) -> None:
+    """Test we handle connection KustoServiceError."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_USER},
+        data=None,
+    )
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["errors"] == {}
+
+    mock_execute_query.side_effect = test_input
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        BASE_CONFIG.copy(),
+    )
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result2["errors"] == {"base": expected}
+
+
+async def test_options_flow(hass, entry_managed) -> None:
+    """Test options flow."""
+    result = await hass.config_entries.options.async_init(entry_managed.entry_id)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "init"
+    assert result["last_step"]
+
```

Comment:
```suggestion
    assert updated["data"] != UPDATE_OPTIONS
```

Reply:
Cant assert this here, the "updated" doesn't exist before the next line?

Reply:
Please check that the entry has different options before updating to make sure the updating works